casanovo/casanovo.py

```
@@ -694,17 +698,33 @@ def _get_weights_from_url(
+            logger.info(
+                "Model weights %s retrieved from local cache (fresh within "
+                "%ds TTL)",
+                file_url,
+                CACHE_TTL_SECONDS,
+            )
+            return cache_file_path
+
+        url_last_modified = time.time()
 
         try:
             file_response = requests.head(file_url)
```

Ruff (0.15.9)

[error] Probable use of `requests` call without timeout

(S113)